mlflow/gateway/providers/anthropic.py

```diff
@@ -20,19 +24,24 @@ async def completions(self, payload: completions.RequestPayload) -> completions.
         payload = jsonable_encoder(payload, exclude_none=True)
         if "top_p" in payload:
             raise HTTPException(
-                status_code=400,
+                status_code=422,
                 detail="Cannot set both 'temperature' and 'top_p' parameters. "
                 "Please use only the temperature parameter for your query.",
             )
-        if "max_tokens" not in payload:
+        max_tokens = payload.get("max_tokens", MLFLOW_AI_GATEWAY_ANTHROPIC_DEFAULT_MAX_TOKENS)
+
+        if max_tokens > MLFLOW_AI_GATEWAY_ANTHROPIC_MAXIMUM_MAX_TOKENS:
             raise HTTPException(
-                status_code=400,
-                detail="You must set an integer value for 'max_tokens' for the Anthropic provider "
-                "that provides the upper bound on the returned token count.",
+                status_code=422,
+                detail="Invalid value for max_tokens: cannot exceed "
+                f"{MLFLOW_AI_GATEWAY_ANTHROPIC_MAXIMUM_MAX_TOKENS}.",
             )
+
+        payload["max_tokens"] = max_tokens
+
         if payload.get("stream", None) == "true":
             raise HTTPException(
-                status_code=400,
+                status_code=422,
```

Comment:
I think 400 is more appropriate for the error codes here based on https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422 and https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400. We'll also use 400 on Databricks.

Reply:
Edit: Nevermind, I like 422. Let's stick with it